### PR TITLE
383 164 authentication updates 9

### PR DESCRIPTION
**Pull request by Firstyear**
This adds the majority of the support for Passkeys, puts in foundations for DeviceKeys (Attested Webauthn), and also removes some older credential update paths. 

- [ x ] cargo fmt has been run
- [ ] cargo clippy has been run
- [ x ] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

**Comment by Firstyear** [comment]
@yaleman I'll do a "self review" first tomorrow :)

**Comment by Firstyear** [comment]
@yaleman I think it's good to review now!

File: kanidmd_web_ui/src/credential/reset.rs [hunk 1]
```diff
@@ -326,23 +349,22 @@ impl CredentialResetApp {
...
                     </button>
                   </>
                 }
             }
             Some(CredentialDetail {
                 uuid: _,
-                claims: _,
                 type_:
                     // TODO: review this and find out why we aren't using these variables
+                    // Because I'm lazy 🙃
```
Comment (yaleman) [inline kanidmd_web_ui/src/credential/reset.rs:367]: 😄

**Review by yaleman — COMMENTED** [review]
- if the domain name / origin are mismatched the only error the users are seeing is in the JS console, it just sits and spins - I'll raise this as a new issue since it's not really a blocker. Folks should configure things properly.

- currently if a user has a password and they add a passkey they get blocked on login with an "unimplemented" message (login.rs#653) this feels like a logic fail somewhere, since users used to be prompted for MFA options?

**Comment by Firstyear** [comment]
I think I know what's missing in login.rs, will add meow.

**Comment by Firstyear** [comment]
All fixed, need to make that domain name thing nicer though as you mentioned.

**Comment by yaleman** [comment]
That's awesome, I like very much.

**Pull request merged**